packages/google_identity_services_web/lib/src/js_interop/shared.dart

```diff
@@ -3,11 +3,18 @@
 // found in the LICENSE file.
 
 /// Attempts to retrieve an enum value from [haystack] if [needle] is not null.
+///
+/// Returns `null` if no enum value in [haystack] matches [needle].
 T? maybeEnum<T extends Enum>(String? needle, List<T> haystack) {
   if (needle == null) {
     return null;
   }
-  return haystack.byName(needle);
+  for (final T value in haystack) {
+    if (value.name == needle) {
+      return value;
+    }
+  }
```

Comment:
A future improvement may be to log a warning or similar when we attempt to decode a value of T that will return null, so people can report bugs. We'll see if that becomes a problem later.